### PR TITLE
Fix issue 106 by adding developer docs

### PR DESCRIPTION
**Pull request by ayusht2810**
#### Summary
Added developer docs in the Readme

#### Ticket Link
#106

### CODE REVIEW

File: docs/developer_docs.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+There are two methods by which you can connect your Mattermost account to your MS Teams account.
+
+- **Using slash command**
+    - Run the slash command `/msteams-sync connect` in any channel.
+    - You will get an ephemeral message from the MS Teams bot containing a link and a code to connect your account.
+    - Click on that link and enter the code. If it asks for login, enter your Microsoft credentials and click `Continue` to authorize and connect your account.
+
+- **Using the button in the full screen modal**
+    - If the setting "Enforce connected accounts" is enabled in the plugin's config settings, then a full screen modal appears that looks like this -
```
Comment (jespino) [inline docs/developer_docs.md:74]: Same here

File: docs/developer_docs.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    - If the setting "Enforce connected accounts" is enabled in the plugin's config settings, then a full screen modal appears that looks like this - 
+    
+    ![image](https://user-images.githubusercontent.com/77336594/226347884-a6469d95-de68-4706-a145-9511d42bd7a4.png)
+
+    - Click on the "Connect account" button. If it asks for login, enter your Microsoft credentials and click `Continue` to authorize and connect your account.
+
+    After connecting successfully, you will get an ephemeral message from the MS Teams bot saying "Your account has been connected".
+
+- **Connecting the bot account**
+    - Run the slash command `/msteams-sync connect-bot` in any channel.
```
Comment (jespino) [inline docs/developer_docs.md:83]: Same here

**Review by jespino — CHANGES_REQUESTED** [review]
There are some information that is out of date after the last changes in authentication and the client library. Other than that it looks good to me.

**Comment by ayusht2810** [comment]
@jespino did the review fixes. Please re-review

**Review by jespino — APPROVED** [review]
LGTM

**Pull request merged**